openid/src/main/java/fish/payara/security/openid/domain/OpenIdContextImpl.java

```diff
@@ -162,14 +164,41 @@ public void setExpiresIn(Long expiresIn) {
     public JsonObject getClaimsJson() {
         if (claims == null) {
             if (configuration != null && accessToken != null) {
-                claims = userInfoController.getUserInfo(configuration, accessToken);
+                if(!configuration.isClaimsFromIDToken()) {
+                    claims = userInfoController.getUserInfo(configuration, accessToken);
+                } else {
+                    LOGGER.log(FINEST, "Processing user info from id token");
+                    claims = processClaimsFromIDToken();
+                }
             } else {
                 claims = Json.createObjectBuilder().build();
             }
         }
         return claims;
     }
 
+    /**
+     * Method to get user information from Id Token
+     * @return JsonObject with user information
+     */
+    private JsonObject processClaimsFromIDToken() {
+        JwtClaims identityTokenJWTClaims = identityToken.getJwtClaims();
+        JwtClaims accessTokenJWTClaims = accessToken.getJwtClaims();
+        //setting profile claims from id token
+        JsonObject userInfo = Json.createObjectBuilder()
+                .add(OpenIdConstant.SUBJECT_IDENTIFIER, identityTokenJWTClaims.getStringClaim(OpenIdConstant.SUBJECT_IDENTIFIER).orElse(""))
+                .add(OpenIdConstant.NAME, identityTokenJWTClaims.getStringClaim(OpenIdConstant.NAME).orElse(""))
+                .add(OpenIdConstant.FAMILY_NAME, accessTokenJWTClaims.getStringClaim(OpenIdConstant.FAMILY_NAME).orElse(""))
+                .add(OpenIdConstant.GIVEN_NAME, accessTokenJWTClaims.getStringClaim(OpenIdConstant.FAMILY_NAME).orElse(""))
```

Comment:
Probably a copy-error, you obviously meant GIVEN_NAME in getStringClaim

Reply:
I already fix it, thanks